arq/jobs.py

```diff
@@ -5,12 +5,35 @@
 from datetime import datetime
 from enum import Enum
 from typing import Any, Callable, Dict, Optional, Tuple
-
+import io
+import builtins
 from aioredis import Redis
 
 from .constants import default_queue_name, in_progress_key_prefix, job_key_prefix, result_key_prefix
 from .utils import ms_to_datetime, poll, timestamp_ms
 
+safe_builtins = {
+    'range',
+    'complex',
+    'set',
+    'frozenset',
+
+    'slice',
+}
+class RestrictedUnpickler(pickle.Unpickler):
+
+    def find_class(self, module, name):
+        """Only allow safe classes from builtins"""
+        if module == "builtins" and name in safe_builtins:
```

Comment:
I don't see how this blocks access to `subprocess` or thousands of other similar risks.